app/controllers/provider_interface/provider_relationship_permissions_controller.rb

```diff
@@ -0,0 +1,67 @@
+module ProviderInterface
+  class ProviderRelationshipPermissionsController < ProviderInterfaceController
+    before_action :render_404_unless_permissions_found
+    before_action :render_403_unless_access_permitted
+
+    def edit
+      initialize_form
+    end
+
+    def update
+      initialize_form
+
+      if @form.update!(provider_relationship_permissions_form_params)
+        flash[:success] = 'Permissions successfully changed'
+        redirect_to provider_interface_organisation_path(provider_relationship_permissions.training_provider)
+      else
+        flash[:warning] = 'Unable to save permissions, please try again. If problems persist please contact support.'
+        render :edit
+      end
+    end
+
+  private
+
+    def initialize_form
+      @form = ProviderRelationshipPermissionsForm.new(permissions: provider_relationship_permissions)
+    end
+
+    def provider_relationship_permissions
+      @provider_relationship_permissions ||= ProviderRelationshipPermissions.find_by(provider_relationship_params)
+    end
+
+    def provider_relationship_params
+      params.permit(:ratifying_provider_id, :training_provider_id).to_h
```

Comment:
Given that all these relationships will exist in the database anyway, would it be simpler to use `:relationship_id`?

Reply:
Yes I think so, the only actions that flow into this controller logic will come from the organisations pages which already lookup permissions records or from the setup redirect which have to retrieve these records anyway.

Reply:
@duncanjbrown https://github.com/DFE-Digital/apply-for-teacher-training/pull/2547/commits/45c09755185c84227644894e4aa87ab68a624558 should cover routing to these actions via the permissions ID.